contrib/babelfishpg_tsql/src/pltsql_coerce.c

```diff
@@ -394,7 +398,15 @@ static CoercionPathType tsql_find_coercion_pathway(Oid sourceTypeId, Oid targetT
 			type_nsoid = typtup->typnamespace;
 			type_nsname = get_namespace_name(type_nsoid);
 			type_name = NameStr(typtup->typname);
-			
+
+			/* We've found INT8 to MONEY casting */
+			if(isInt8Type && strcmp(type_nsname, "sys") == 0 && ((strcmp(type_name, "money") == 0) || (strcmp(type_name, "smallmoney") == 0)))
+     			isInt8ToMoney = true;
+
+			/* Check if sourceType is INT8 */
```

Comment:
This comment is not actually true. `isInt8Type` can be true if targetType is int8.